maps/oldfare/jobs/soldiers.dm

```diff
@@ -266,4 +266,11 @@ GLOBAL_LIST_INIT(lone_thoughts, list(
 		return
 	var/datum/quirk/chosen_quirk = pick(random_quirks)
 	src.quirk = new chosen_quirk
-	to_chat(src, "<span class='bnotice'>I was formed a bit different. I am [quirk.name]. [quirk.description]</span>")
\ No newline at end of file
+	to_chat(src, "<span class='bnotice'>I was formed a bit different. I am [quirk.name]. [quirk.description]</span>")
+	switch(chosen_quirk)
+		if(/datum/quirk/cig_addict)
```

Comment:
Drop the switch and just make these explicit istype checks isntead.
```suggestion
		if(istype(chosen_quirk, /datum/quirk/cig_addict))
```

Reply:
For some reason, istype does not work in this case, in the start I made it like that but the check would not be true even with the /datum/quirk/cig_addict hardcoded in chosen quirk.

Reply:
Oh? So this method does work for sure then?

Reply:
For as much as I could test yes. I even put some admin logs to see and only after I changed it from istype to a normal if, I would receive the admin log.

Reply:
I will test it again.. one moment

Reply:
It works